public/content/posts/performance-improvement-in-a-next.js-page/index.md

```diff
@@ -0,0 +1,121 @@
+imageSrc: performance.png
+imageAlt: Performance score in lighthouse
+author: gayan-witharana
+tags:
+  - performance
+  - web-development
+  - Next.js
+  - web-vitals
+---
+
+## Are web vitals necessary?
+
+From Google monthly user experience data, between May and October 2021, overall Core web vitals compliance increased by 58 per cent worldwide.
+
+- Largest Contentful Paint – 44 per cent compliance
+- Cumulative Layout Shift – 71 per cent met requirements
+- First Input Delay – 91 per cent compliance
+
+Reference: https://www.figmentagency.com/seo/core-web-vitals-seo-updates-for-2022
+
+So obviously if you're not optimizing for core web vitals, you are definitely in the minority.
+
+## LCP
+
+In the Fishbrain web application, we're doing fine with the FID and CLS scores as you can see in the title image. But the LCP score is quite hight compared to the standard time (2.5s).
+
+![LCP score margin](/lcp.png "LCP Score")
+
+### Identifying the LCP of a page
+
+There are few ways we can identify the LCP of a page. One easy way is to use the Performance Insights tab in Google Chrome. Once you click the "Measure page load" button it will run for a few seconds and then it will show some details related to performance. On the right side of the tab, you will be able to see web vitals and once you hover LCP it will highlight the LCP content of the page.
+
+![Google Chrome Performance Insights](/perf-insight.png "Google Chrome Performance Insights")
+
+### Improve the performance of an image
+
+Once we identify the LCP element of the page, we can try to make sure it loads relatively fast. Since in this case it's an image we can load the image in "eager" mode instead of the "lazy" mode.
+
+```html
+<img src="imageUrl" loading="eager" />
+```
+
+Since `eager` is the default value for the loading attribute, in this case we can remove the loading prop as well.
+
+[This Article](/posts/improving-images-on-fishbrain.com) from our blog also explains how we have improved the overall performance of the images.
+
+### Preload the LCP image
+
+Preloading the image inside next/head is another effective way to improve the LCP score.
+
+```jsx
+<Head>
+  <link rel="preload" href={imageUrl} as="image" />
+</Head>
+```
+
+This will ask the browser to load image upfront even before it discovers the image element.
+
+### Next.js Dynamic imports
+
+For testing a single page (url) performance, [web.dev/measure](https://web.dev/measure) is a great tool. After measuring the performance with this tool, it provides us some useful methods on how to improve our web vitals scores.
+One of method it suggests is to reduce the initial response time. For this we could use [Next.js dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import). It's a handy way to load components and libraries only when those are needed. Therefore we could use the dynamic imports when we are loading components conditionally. This will also slightly reduce the build size.
```

Comment:
```suggestion
One of the methods it suggests is to reduce the initial response time. For this we could use [Next.js dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import). It's a handy way to load components and libraries only when those are needed. Therefore we could use the dynamic imports when we are loading components conditionally. This will also slightly reduce the build size.
```